Pinta.Core/Classes/AsyncEffectRenderer.cs

```diff
@@ -264,16 +262,19 @@ void RenderTile (int renderId, int threadId, int tileIndex)
 		// Update bounds to be shown on next expose.
 		lock (updated_lock) {
 			if (is_updated) {
-				updated_x1 = Math.Min (bounds.X, updated_x1);
-				updated_y1 = Math.Min (bounds.Y, updated_y1);
-				updated_x2 = Math.Max (bounds.X + bounds.Width, updated_x2);
-				updated_y2 = Math.Max (bounds.Y + bounds.Height, updated_y2);
+				updated_area = RectangleI.FromLTRB (
+					left: Math.Min (bounds.X, updated_area.X),
```

Comment:
I think this is now just equivalent to `RectangleI.Union()`?

Reply:
Thanks, I didn't know this method already existed. I've updated the code accordingly